modules/roles/rest/handlers.go

```diff
@@ -0,0 +1,74 @@
+package rest
+
+import (
+	"encoding/hex"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	abci "github.com/tendermint/abci/types"
+	"github.com/tendermint/basecoin"
+	"github.com/tendermint/basecoin/client/commands"
+	"github.com/tendermint/basecoin/errors"
+	"github.com/tendermint/basecoin/modules/base"
+	"github.com/tendermint/basecoin/modules/nonce"
+	"github.com/tendermint/basecoin/modules/roles"
+	"github.com/tendermint/tmlibs/common"
+)
+
+type RoleInput struct {
+	Role string `json:"role" validate:"required,min=2"`
+
+	MinimumSigners uint32 `json:"min_sigs" validate:"required,min=1"`
+
+	Signers []basecoin.Actor `json:"signers" validate:"required,min=1"`
+
+	Sequence uint32 `json:"seq" validate:"required,min=1"`
+}
+
+func parseRole(roleInHex string) ([]byte, error) {
+	parsedRole, err := hex.DecodeString(common.StripHex(roleInHex))
+	if err != nil {
+		err = errors.WithMessage("invalid hex", err, abci.CodeType_EncodingError)
+		return nil, err
+	}
+	return parsedRole, nil
+}
+
+// mux.Router registrars
+
+// RegisterQueryAccount is a mux.Router handler that exposes GET
```

Comment:
This comment appears stale. Should be `RegisterCreateRole` and route is `/build/create_role`